test/unit/dirty_associations_test.rb

```diff
@@ -1,6 +1,7 @@
 require 'test_helper'
 
-class DummyDirtyAssociationsModel
+class DummyDirtyAssociationsModel < ApplicationRecord
```

Comment:
I think it's good to have an example tested that doesn't extend ApplicationRecord to show that the existing behavior doesn't actually require a real database model. Is it necessary to change this?

Reply:
If ApplicationRecord is removed, DummyDirtyAssociationsModel becomes a class and after_save will not work on it.

Reply:
you can implement/mock after_save on plain class too

Reply:
I think we should preserve the original behavior which can be done like Marek mentioned or conditionally defining the after_save callback if the class actually responds to such a method

Reply:
@sseelam2 this concern does not seem incorporated, do you plan to change it?

Reply:
@ares Yes. I'm on it.